src/ui/pages/SwapForm/SwapForm.tsx

```diff
@@ -956,10 +961,51 @@ export function SwapFormComponent() {
   );
 }
 
+/** Sets initial chainInput to last used chain for current address */
+function SwapFormPrepareChain({ children }: React.PropsWithChildren) {
+  const { singleAddress: address, ready } = useAddressParams();
+  const navigationType = useNavigationType();
+  const isBackOrForward = navigationType === 'POP';
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [prepared, setPrepared] = useState(
+    isBackOrForward || searchParams.has('chainInput')
+  );
+  const { data: lastUsedChain, isFetchedAfterMount } = useQuery({
+    enabled: ready && !prepared,
+    // Avoid using stale value. Leaving form and coming back should use new value instantly
+    staleTime: 0,
+    queryKey: ['wallet/getLastSwapChainByAddress', address],
+    queryFn: () => walletPort.request('getLastSwapChainByAddress', { address }),
+  });
+  useEffect(() => {
+    if (prepared || !isFetchedAfterMount) {
+      return;
+    }
+    if (lastUsedChain) {
+      searchParams.set('chainInput', lastUsedChain);
+      setSearchParams(searchParams, { replace: true });
```

Comment:
With this approach we will keep this chain for other addresses after changing them.
In this case user did nothing to set up the chain manually, so may be it will be better to select another default chain for other address?

Reply:
discussed)